### ISSUE TITLE
Cannot access the content of `wt_opt` in examples/02_references_turbines

### ISSUE THREAD

**Issue by ThomasPotentier** [WISDEM/WISDEM#576]
## Description
Upon installing WISDEM following the readme, I tried to launch examples\02_reference_turbines\nrel5mw_driver.py but encountered the following error:

![image](https://github.com/user-attachments/assets/da0c929a-c905-47d9-aa27-82cb44dbcfc9)

Similarly, following the example in https://wisdem.readthedocs.io/en/master/examples/02_refturb/tutorial.html, trying to copy/paste the script `sample_plot.py` leads to a similar error:

![image](https://github.com/user-attachments/assets/ac3d2b89-86b6-457c-ba43-b5dbc4c8d474)

Digging inside `refturb`, it is possible to access up to "one level up" to the non dimensional span by calling a slighlty different name compared to the example provided: 

`refturb.model.wt.wt_prop.wt_init.blade.outer_shape_bem.compute_blade_outer_shape_bem` instead of `refturb["wt.wt_init.blade.outer_shape_bem.compute_blade_outer_shape_bem.s_default"]`. However as the screenshot below shows, there are no attribute containing the missing `s_default` values as in the example. 

![image](https://github.com/user-attachments/assets/c6b6a689-f0ce-4a01-9766-b40e5fc905b7)

Unfortunately, it is an issue for all varaible. It feels like data handling has changed in OpenMDAO ?

Despite the error message, the process seems to export correctly (part of) the data:

![image](https://github.com/user-attachments/assets/32fdd325-c66a-4745-9a9b-3e693e1835db)

### Steps to reproduce issue

1. Install WISDEM 
2. Run examples\02_reference_turbines\nrel5mw_driver.py
3. Run https://wisdem.readthedocs.io/en/master/examples/02_refturb/tutorial.html, `sample_plot.py`

### Code versions
Below are the packges list and version
alabaster==1.0.0
archspec==0.2.3
attrs==24.2.0
Babel==2.14.0
boltons==24.0.0
Brotli==1.1.0
certifi==2024.8.30
cffi==1.17.1
charset-normalizer==3.4.0
colorama==0.4.6
conda==24.9.2
conda-libmamba-solver==24.9.0
conda-package-handling==2.4.0
conda_package_streaming==0.11.0
contourpy==1.3.0
coverage==7.6.3
coveralls==4.0.1
cycler==0.12.1
Cython==3.0.11
dearpygui==2.0.0
distro==1.9.0
docopt==0.6.2
docutils==0.21.2
et-xmlfile==1.1.0
exceptiongroup==1.2.2
fonttools==4.54.1
frozendict==2.4.6
h2==4.1.0
hpack==4.0.0
hyperframe==6.0.1
idna==3.10
imagesize==1.4.1
importlib_resources==6.4.5
iniconfig==2.0.0
Jinja2==3.1.4
jsonpatch==1.33
jsonpointer==3.0.0
jsonschema==4.23.0
jsonschema-specifications==2024.10.1
kiwisolver==1.4.7
libmambapy==1.5.10
MarkupSafe==3.0.1
marmot-agents==0.2.5
matplotlib==3.9.2
mdolab-baseclasses==1.8.0
menuinst==2.1.2
meson==1.5.2
MoorPy==1.1.0
mpi==1.0.0
mpi4py==4.0.1
munkres==1.1.4
networkx==3.4.1
nlopt==2.8.0
numpy==1.26.4
numpydoc==1.8.0
openmdao==3.35.0
openpyxl==3.1.5
packaging==24.1
pandas==2.2.3
patsy==0.5.6
pillow==10.4.0
pip==24.2
pkgutil_resolve_name==1.3.10
platformdirs==4.3.6
pluggy==1.5.0
pycosat==0.6.6
pycparser==2.22
pyDOE3==1.0.4
Pygments==2.18.0
pyoptsparse==2.11.3
pyparsing==3.2.0
PySide6==6.8.0
PySocks==1.7.1
pytest==8.3.3
pytest-cov==5.0.0
python-benedict==0.33.2
python-dateutil==2.9.0
python-fsutil==0.14.1
python-slugify==8.0.4
pytz==2024.1
PyYAML==6.0.2
referencing==0.35.1
requests==2.32.3
rpds-py==0.20.0
ruamel.yaml==0.18.6
ruamel.yaml.clib==0.2.8
scipy==1.14.1
setuptools==75.1.0
shiboken6==6.8.0
simpy==4.1.1
six==1.16.0
snowballstemmer==2.2.0
sortedcontainers==2.4.0
Sphinx==8.1.3
sphinxcontrib-applehelp==2.0.0
sphinxcontrib-devhelp==2.0.0
sphinxcontrib-htmlhelp==2.1.0
sphinxcontrib-jsmath==1.0.1
sphinxcontrib-qthelp==2.0.0
sphinxcontrib-serializinghtml==1.1.10
sqlitedict==2.1.0
statsmodels==0.14.4
tabulate==0.9.0
text-unidecode==1.3
toml==0.10.2
tomli==2.0.2
tornado==6.4.1
tqdm==4.66.5
truststore==0.9.2
tzdata==2024.2
urllib3==2.2.3
wheel==0.44.0
win_inet_pton==1.1.0
wisdem==3.17.0
zipp==3.20.2
zstandard==0.23.0

**Comment by gbarter** [comment]
Hello,

Thank you for bringing this to our attention.  For the errors you spotted:

- The `sample_plot.py` was 100% out of date.  I have fixed it in the `develop` branch [here](https://github.com/WISDEM/WISDEM/blob/develop/examples/02_reference_turbines/sample_plot.py).  If you are comfortable moving to that branch, you can do so (`git checkout develop; git pull`), or you can copy the new code from the link if you installed WISDEM as a library (`conda install wisdem`).
- The error you say in the `nrel5mw_driver.py` script is more surprising and one I cannot replicate either by installing from source or as a conda package.  That code is also more regularly tested.  Make sure you are consistent with the latest version [here](https://github.com/WISDEM/WISDEM/blob/master/examples/02_reference_turbines/nrel5mw_driver.py).

**Comment by ThomasPotentier** [comment]
Hello Garrett,

Thanks for updating `sample_plot.py`, it is now running. However the outcome is different from the expected values:

![image](https://github.com/user-attachments/assets/47a23412-838d-4455-ba1f-6d12c3dd9c10)


As you can see, there is a dip in the induction at 2% span and towards the tip it goes up to a = 0.5 . Opposite to the plot provided in the doc, where the behaviour is more common:

![image](https://github.com/user-attachments/assets/21dca1c5-81ea-4543-aa34-22b2cef531ec)


Similarly, the expected outcome by running the 5MW turbine is different from the documentation:
mine:
![image](https://github.com/user-attachments/assets/829f7518-6b79-4049-869a-d84171b599df)
doc:
![image](https://github.com/user-attachments/assets/0efaf150-1a43-486a-a908-78471e088332)


Differences are quite small but I am surprised to see any for a reference case.


Regarding the second error, despite removing WISDEM completely and re-installing it it still have the same issue. I have also ensured to use the script you provided in your previous answer.

However, by using `load_wisdem("outputs/refturb_output.pkl")`, I am able to reload the outputs (in a similar fashion used in `sample_plot.py`). In that case I am now able to access the desired variables :

![image](https://github.com/user-attachments/assets/1f4bcd2f-a8b6-440c-b622-918a674c2356)
![image](https://github.com/user-attachments/assets/d42824de-01c1-4e6f-ba78-3928d6941132)


It may be a work around at the moment, but hopefully I won't have to reload the pickled results before analysing them for othger blades!

**Comment by gbarter** [comment]
Apologies for the delayed reply. . .

The graph shown in the documentation [here](https://wisdem.readthedocs.io/en/master/examples/02_refturb/tutorial.html) is from the IEA Wind 15-MW example.  When I run `iea15mw_driver.py` and then `sample_plot.py`, I get a profile that still has a kink towards the root and has an induction factor that tails off at the root, as expected.  Nevertheless, you raise some good points and I will look into the underlying cause.

On the differences in AEP, LCOE, blade deflection from the documentation numbers to now, that is just due to freezing the documentation a long time away and not updating it as WISDEM has evolved and improved.  I have put in an edit to the documentation to make it more current.

On the errors you are still getting when just trying to pull up the blade mass in the python script, for that I am still not positive what is going on.  One possibility is the OpenMDAO library on your installation is more particular about the variable naming than I am seeing.  What happens if you try `tcc.blade_mass` instead of `wt.tcc.blade_mass` in the script (essentially remove the `wt.` string from the output names)?

**Comment by ThomasPotentier** [comment]
Hello Garrett,

After running the IEA 15MW I obtain the same plot as in the documentation. Glad to see I was comparing two different turbines.

Also, it is reassuring to see that the CCBlade differences noted are simply an out of date reference screenshot!

Finally, the first time I installed WISDEM, I did try removing `wt` from the output names to no avail. Howver, since I reinstalled it, removing `wt` from the variable name is working. I think we can close the issue !

Thanks for your speedy replies and answers found!

Best,

Thomas
